hack/tools/update_providers/provider_issues.go

```diff
@@ -0,0 +1,210 @@
+		"kubernetes-sigs/cluster-api-provider-ibmcloud",
+		"kubernetes-sigs/cluster-api-provider-nested",
+		"oracle/cluster-api-provider-oci",
+		"kubernetes-sigs/cluster-api-provider-openstack",
+		"kubernetes-sigs/cluster-api-operator",
+		"kubernetes-sigs/cluster-api-provider-packet",
+		"kubernetes-sigs/cluster-api-provider-vsphere",
+	}
+)
+
+// Issue is the struct for the issue.
+type Issue struct {
+	// Title is the title of the issue.
+	Title string `json:"title"`
+
+	// Body is the body of the issue.
+	Body string `json:"body"`
+}
+
+// IssueResponse is the struct for the issue response.
+type IssueResponse struct {
+	// HTMLURL is the URL of the issue.
+	HTMLURL string `json:"html_url"`
+}
+
+func main() {
+	token := flag.String("token", "", "GitHub personal access token")
+	dryRun := flag.Bool("dryRun", true, "Dry run")
+	flag.Parse()
+
+	if *token == "" {
+		fmt.Println("GitHub personal access token is required.")
+		os.Exit(1)
+	}
+
+	issueBody := `
+<!-- body -->
+<!-- TODO: remove all TODOs before running this utility -->
+<!-- TODO: update CAPI release semver -->
+CAPI v1.x.0-beta.0 has been released and is ready for testing.
+Looking forward to your feedback before CAPI 1.x.0 release!
+
+## For quick reference
+
+<!-- body -->
+<!-- TODO: CAPI release notes -->
+- [CAPI v1.x.0-beta.0 release notes](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.x.0-beta.0)
+- [Shortcut to CAPI git issues](https://github.com/kubernetes-sigs/cluster-api/issues)
+
+## Following are the planned dates for the upcoming releases
+
+<!-- TODO: update CAPI release timeline -->
+|Release|Expected Date|
+|-----|-----|
+|v1.5.0-beta.x | Tuesday 5th July 2023|
+|release-1.5 branch created (Begin [Code Freeze])|Tuesday 11th July 2023|
+|v1.5.0-rc.0 released|Tuesday 11th July 2023|
+| release-1.5 jobs created | Tuesday 11th July 2023 |
+| v1.5.0-rc.x released | Tuesday 18th July 2023 |
+| v1.5.0 released | Tuesday 25th July 2023 |
+<!-- body -->
+
+<!-- [List of CAPI providers](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md#communicate-beta-to-providers) -->
+`
+
+	if *dryRun {
+		fmt.Printf("\n")
+		fmt.Println("###############################################")
+		fmt.Println("Dry run is enabled. No git issues will be created.")
+		fmt.Println("###############################################")
+		fmt.Printf("\n")
+	}
+
+	fmt.Println("\n\nCreating issues for the following repositories:")
+	fmt.Println(strings.Join(repoList, "\n"))
+	fmt.Printf("\n")
+
+	if !*dryRun {
+		fmt.Println("Continue to create issues? (y/n)")
+		var response string
+		_, err := fmt.Scanln(&response)
+		if err != nil {
+			fmt.Printf("Failed to read response: %s\n", err)
+			os.Exit(1)
+		}
+		if response != "y" {
+			fmt.Println("Aborting...")
+			os.Exit(0)
+		}
+	}
+	fmt.Printf("\n")
+
+	for _, repo := range repoList {
+		url := fmt.Sprintf("%s/repos/%s/issues", baseURL, repo)
+		if *dryRun {
```

Comment:
It would be really useful if `dryRun` also printed the full templated issue that would be created to the output.

Reply:
Updated.